src/MOI_wrapper.jl

```diff
@@ -0,0 +1,414 @@
+
+    @SC SCIPchgLhsLinear(scip, cons, lhs)
+    @SC SCIPchgRhsLinear(scip, cons, rhs)
+
+    return nothing
+end
+
+function MOI.is_valid(o::Optimizer, ci::CI{SAF,<:BOUNDS})
+    1 <= ci.value <= length(o.mscip.cons)
+end
+
+function MOI.get(o::Optimizer, ::MOI.NumberOfConstraints{F,S}) where {F,S}
+    haskey(o.constypes, (F, S)) ? length(o.constypes[F, S]) : 0
+end
+
+function MOI.get(o::Optimizer, ::MOI.ConstraintFunction, ci::CI{SVF, S}) where S <: BOUNDS
+    SVF(ci)
+end
+
+function MOI.get(o::Optimizer, ::MOI.ConstraintSet, ci::CI{SVF, S}) where S <: BOUNDS
+    var = get_var(o.mscip, ci.value)
+    lb, ub = SCIPvarGetLbOriginal(var), SCIPvarGetUbOriginal(var)
+    from_bounds(S, lb, ub)
+end
+
+function MOI.get(o::Optimizer, ::MOI.ConstraintFunction, ci::CI{SAF, S}) where S <: BOUNDS
+    scip, cons = get_scip(o), get_cons(o, ci)
+    nvars::Int = SCIPgetNVarsLinear(scip, cons)
+    vars = unsafe_wrap(Array{Ptr{SCIP_VAR}}, SCIPgetVarsLinear(scip, cons), nvars)
+    vals = unsafe_wrap(Array{Float64}, SCIPgetValsLinear(scip, cons), nvars)
+
+    terms = [SAT(vals[i], VI(get_index(o, vars[i])))
+             for i=1:nvars]
+    # can not identify constant anymore (is merged with lhs,rhs)
+    return SAF(terms, 0.0)
+end
+
+function MOI.get(o::Optimizer, ::MOI.ConstraintSet, ci::CI{SAF, S}) where S <: BOUNDS
+    scip, cons = get_scip(o), get_cons(o, ci)
+    lhs, rhs = SCIPgetLhsLinear(scip, cons), SCIPgetRhsLinear(scip, cons)
+    from_bounds(S, lhs, rhs)
+end
+
+function MOI.get(o::Optimizer, ::MOI.ConstraintName, ci::CI{SAF,<:BOUNDS})
+    SCIPconsGetName(get_cons(o, ci))
+end
+
+function MOI.set(o::Optimizer, ::MOI.ConstraintName, ci::CI{SAF,<:BOUNDS}, name::String)
+    @SC SCIPchgConsName(get_scip(o), get_cons(o, ci), name)
+end
+
+function MOI.set(o::Optimizer, ::MOI.ObjectiveFunction{SAF}, obj::SAF)
+    allow_modification(o)
+    scip = get_scip(o)
+
+    # reset objective coefficient of all variables first
+    for v in o.mscip.vars
+        @SC SCIPchgVarObj(scip, v[], 0.0)
+    end
+
+    # set new objective coefficients, summing coefficients
+    for t in obj.terms
+        var = get_var(o, t.variable_index)
+        oldcoef = SCIPvarGetObj(var)
+        newcoef = oldcoef + t.coefficient
+        @SC SCIPchgVarObj(scip, var, newcoef)
+    end
+
+    @SC SCIPaddOrigObjoffset(scip, obj.constant - SCIPgetOrigObjoffset(scip))
+
+    return nothing
+end
+
+function MOI.get(o::Optimizer, ::MOI.ObjectiveFunction{SAF})
+    terms = SAT[]
+    for i = 1:length(o.mscip.vars)
+        vi = VI(i)
+        coef = SCIPvarGetObj(get_var(o, vi))
+        coef == 0.0 || push!(terms, SAT(coef, vi))
+    end
+    constant = SCIPgetOrigObjoffset(get_scip(o))
+    return SAF(terms, constant)
+end
+
+function MOI.set(o::Optimizer, ::MOI.ObjectiveSense, sense::MOI.OptimizationSense)
+    allow_modification(o)
+    if sense == MOI.MIN_SENSE
+        @SC SCIPsetObjsense(get_scip(o), SCIP_OBJSENSE_MINIMIZE)
+    elseif sense == MOI.MAX_SENSE
```

Comment:
`FEASIBILITY_SENSE` should also be handled, or at least note that nothing needs to be done.

Reply:
Hm, we could reset the objective coefficients, I guess. But then they could not be recovered easily.

Reply:
Added a warning, and opened issue #80.